### DISCUSSION TITLE
Bypassing the random delay when testing

### DISCUSSION THREAD

**Discussion by hollymcr** [go-acme/lego#1875] [Q&A] [answered]
I have a script which I run from cron. Although I always make a point of picking random start times I fully accept the logic behind the random delay in these scenarios.

But when something isn't working and I need to test my script at the commandline, the random delay is a major pain!

My script redirects lego's output to a logfile, so although (AIUI) lego tries to detect the difference between automated and non-automated runs depending on whether STDOUT is a TTY, clearly lego doesn't recognise my scenario as non-automated.

What is the correct way for me to handle this?

**Comment by ldez** [comment, marked as answer]
Hello,

you can use the flag `--no-random-sleep`

https://go-acme.github.io/lego/usage/cli/options/

**Reply by hollymcr** [reply]
I'm getting:
`Incorrect Usage. flag provided but not defined: -no-random-sleep`
Note that I am passing --no-random-sleep (not -no-random-sleep)

lego 4.10.2

**Reply by ldez** [reply]
it's a flag for the renew command.
```
lego renew --no-random-sleep
```

**Reply by hollymcr** [reply]
> it's a flag for the renew command.

Awesome! I'd found the flag in the source code but never got it working, and the docs don't seem to mention it (including the link to the CLI opts you shared earlier, unless I'm missing something).

**Reply by ldez** [reply]
the flag is inside the tag `lego help renew`  https://go-acme.github.io/lego/usage/cli/options/

**Reply by hollymcr** [reply]
> the flag is inside the tag `lego help renew` https://go-acme.github.io/lego/usage/cli/options/

I don't know how many times I've looked at that page and never noticed the multiple tabs! (I even tried searching the site for no-random-sleep, but I have now discovered that you can't just paste what you're looking for into the search box, you have to type it for the AJAX querying to pick it up.

User error!
